DasharoPayloadPkg/BlSupportDxe/BlSupportDxe.c

```diff
@@ -273,6 +278,25 @@ BlDxeEntryPoint (
     ASSERT_EFI_ERROR (Status);
   }
 
+  Status = ParseFwInfo (&FwGuid, &FwVersion, &FwLsv);
+  if (!EFI_ERROR (Status)) {
+    Esrt = AllocateZeroPool (sizeof (EFI_SYSTEM_RESOURCE_TABLE) + sizeof (EFI_SYSTEM_RESOURCE_ENTRY));
+    ASSERT (Esrt != NULL);
+
+    Esrt->FwResourceVersion  = EFI_SYSTEM_RESOURCE_TABLE_FIRMWARE_RESOURCE_VERSION;
+    Esrt->FwResourceCount    = 1;
+    Esrt->FwResourceCountMax = 1;
+
+    Esre = (EFI_SYSTEM_RESOURCE_ENTRY *)&Esrt[1];
+    CopyMem (&Esre->FwClass, &FwGuid, sizeof (EFI_GUID));
+    Esre->FwType                   = ESRT_FW_TYPE_SYSTEMFIRMWARE;

```

Comment:
Is this something we will want to pass from coreboot as well? Or leave it for FMP to worry about?

Reply:
Possible values are:
* `ESRT_FW_TYPE_UNKNOWN`
* `ESRT_FW_TYPE_SYSTEMFIRMWARE`
* `ESRT_FW_TYPE_DEVICEFIRMWARE`
* `ESRT_FW_TYPE_UEFIDRIVER`

It seems like coreboot won't need anything other than `ESRT_FW_TYPE_SYSTEMFIRMWARE`.

Reply:
I'm not so sure. At some point we may want to track ME version, and if it is disabled by coreboot, edk2 may no longer be able to communicate with it to ask for its version (if this is how it's done, I don't know ME to well). I think it should be of type `DEVICEFIRMWARE`, because even though it is stored on the same flash device as main firmware, it has to be flashed in a different way.

EC also is `DEVICEFIRMWARE`. @mkopec since you've worked with it a lot, do you know if there is a possible scenario in which coreboot is able to obtain EC firmware version, but payload/OS isn't?

Reply:
Separation for capsules type seems to be based more on whether BIOS is locked or not (`ProcessCapsules()` is called first when it's unlocked which is when system firmware and devices without embedded drivers are updated and second time after applying the lock; although this separation is less applicable when coreboot is used) rather than on anything else.

Adding more fields isn't a problem, but type alone probably won't help much because there will be different kinds of ECs.  We'll likely have multiple FMP implementations each supporting a set of GUIDs that it recognizes and knows how to flash with ability to derive necessary information from a GUID.  We should also be able to add new fields describing firmware in the future and detect struct version by the size, or could add a version field from the start.

Reply:
> EC also is DEVICEFIRMWARE. @mkopec since you've worked with it a lot, do you know if there is a possible scenario in which coreboot is able to obtain EC firmware version, but payload/OS isn't?

I don't believe so, right now we don't hide or lock down any EC interfaces so everything should be available to the payload.